.github/workflows/workflow-from-flake-checks.nix

```diff
@@ -0,0 +1,151 @@
+# Runs Nix flake checks individually with group markers for GitHub Actions.
+# Invoke with `nix-shell workflow-from-flake-checks.nix`
+
+let
+  self = import ../..;
+
+  pkgs = self.inputs.nixpkgs.legacyPackages.${builtins.currentSystem};
+  inherit (self.inputs.nixpkgs) lib;
+
+  checkAttrs =
+    lib.foldl'
+      (
+        acc:
+        {
+          name,
+          value,
+        }:
+
+        acc // {
+          "${name}" = {
+            inherit (value) description;
+            platforms = acc.${name}.platforms or {} // { "${value.platform}" = null; };
+            phps = acc.${name}.phps or {} // { "${value.php}" = null; };
+          };
+        }
+      )
+      {}
+      (
+        builtins.concatLists (
+          lib.mapAttrsToList
+            (
+              platform:
+              checks:
+
+              lib.mapAttrsToList
+                (
+                  attr:
+                  check:
+
+                  let
+                    match = builtins.match "(php[0-9]+)-(.+)" attr;
+                  in
+                  {
+                    name = builtins.head (builtins.tail match);
+                    value = {
+                      inherit (check) description;
+                      inherit platform;
+                      php = builtins.head match;
+                    };
+                  }
+                )
+                checks
+            )
+            self.outputs.checks
+        )
+      );
+
+  packages = self.outputs.packages.${builtins.currentSystem};
+  phpPackages = builtins.filter (name: builtins.match "php[0-9]+" name != null) (builtins.attrNames packages);
+
+  workflow = {
+    name = "Test";
+
+    on = {
+      workflow_call = {
+        secrets = {
+          CACHIX_AUTH_TOKEN = {
+            required = true;
+          };
+        };
+      };
+    };
+
+    jobs = {
+      test = {
+        name = "PHP \${{ matrix.php.branch }}";
+        runs-on = "\${{ matrix.os.image }}";
+        steps = [
+          {
+            uses = "actions/checkout@v2";
+          }
+          {
+            name = "Install Nix";
+            uses = "cachix/install-nix-action@v14";
+          }
+          {
+            name = "Set up Nix cache";
+            uses = "cachix/cachix-action@v10";
+            "with" = {
+              authToken = "\${{ secrets.CACHIX_AUTH_TOKEN }}";
+              name = "fossar";
+            };
+          }
+        ]
+        ++ lib.mapAttrsToList
```

Comment:
Or maybe it would be enough to add sort here, that would ensure `php` test is first (not sure if there are any other dependencies – if yes, we should probably express them in `checks.nix` and use toposort).